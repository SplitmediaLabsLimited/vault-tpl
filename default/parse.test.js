const parse = require('./parse');

test('empty string', () => {
  expect(parse(``)).toMatchSnapshot();
});

test('no secrets', () => {
  expect(parse(`hello world`)).toMatchSnapshot();
});

test('1 secret', () => {
  const content = `
  secret: (( vault "version-test:secret@2" ))
  secret2: (( vault "version-test:secret2@3" ))
  secret3: "raw"
  `;
  expect(parse(content)).toMatchSnapshot();
});

test('multiple secrets', () => {
  const content = `
  secret: (( vault "version-test:secret@2" ))
  secret2: (( vault "version-test:secret2@3" ))
  secret3: "raw"
  `;
  expect(parse(content)).toMatchSnapshot();
});

test('multiple and duplicate secrets, should deduplicate', () => {
  const content = `
  secret: (( vault "version-test:secret@2" ))
  secret: (( vault "version-test:secret@2" ))
  secret: (( vault "version-test:secret@2" ))
  secret2: (( vault "version-test:secret2@3" ))
  secret3: "raw"
  `;
  expect(parse(content)).toMatchSnapshot();
});

test('no spaces', () => {
  const content = `
  secret: ((vault "version-test:secret@2"))
  secret: ((vault "version-test:secret@2"))
  secret: ((vault "version-test:secret@2"))
  secret2: ((vault "version-test:secret2@3"))
  secret3: "raw"
  `;
  expect(parse(content)).toMatchSnapshot();
});
