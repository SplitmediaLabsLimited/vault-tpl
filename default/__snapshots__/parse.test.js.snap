// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 secret 1`] = `
Array [
  Object {
    "field": "secret",
    "id": "version-test:secret@2",
    "secret": "version-test",
    "version": "2",
  },
  Object {
    "field": "secret2",
    "id": "version-test:secret2@3",
    "secret": "version-test",
    "version": "3",
  },
]
`;

exports[`empty string 1`] = `Array []`;

exports[`multiple and duplicate secrets, should deduplicate 1`] = `
Array [
  Object {
    "field": "secret",
    "id": "version-test:secret@2",
    "secret": "version-test",
    "version": "2",
  },
  Object {
    "field": "secret2",
    "id": "version-test:secret2@3",
    "secret": "version-test",
    "version": "3",
  },
]
`;

exports[`multiple secrets 1`] = `
Array [
  Object {
    "field": "secret",
    "id": "version-test:secret@2",
    "secret": "version-test",
    "version": "2",
  },
  Object {
    "field": "secret2",
    "id": "version-test:secret2@3",
    "secret": "version-test",
    "version": "3",
  },
]
`;

exports[`no secrets 1`] = `Array []`;

exports[`no spaces 1`] = `
Array [
  Object {
    "field": "secret",
    "id": "version-test:secret@2",
    "secret": "version-test",
    "version": "2",
  },
  Object {
    "field": "secret2",
    "id": "version-test:secret2@3",
    "secret": "version-test",
    "version": "3",
  },
]
`;
